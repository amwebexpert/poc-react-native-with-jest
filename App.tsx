import React from 'react';
import { SafeAreaView, StyleSheet } from 'react-native';
import { DefaultTheme, Provider as PaperProvider, Title, useTheme } from 'react-native-paper';

const theme = {
  ...DefaultTheme,
  colors: {
    ...DefaultTheme.colors,
    primary: 'red',
    accent: 'lightgrey',
  },
};

const App = () => {
  const theme = useTheme();

  return (
    <PaperProvider theme={theme}>
      <SafeAreaView style={styles.container}>
        <Title style={styles.title}>Welcome to unit tested app!</Title>
      </SafeAreaView>
    </PaperProvider>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    color: theme.colors.primary,
    backgroundColor: theme.colors.accent,
    borderWidth: 1,
    borderColor: theme.colors.primary,
    padding: 20,
  },
});

export default App;
