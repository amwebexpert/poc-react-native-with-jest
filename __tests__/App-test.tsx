import React from 'react';
import { render } from '@testing-library/react-native';
import App from '../App';

it('renders correctly', async () => {
    // given
    const { getByText } = render(<App />);

    // when
    const welcome = getByText("Welcome to unit tested app!");

    // then
    expect(welcome).toBeDefined();
});
