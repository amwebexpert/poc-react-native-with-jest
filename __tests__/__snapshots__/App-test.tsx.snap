// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly A 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "500",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 20,
                "letterSpacing": 0.15,
                "lineHeight": 30,
                "marginVertical": 2,
              },
              Object {
                "backgroundColor": "lightgrey",
                "borderColor": "red",
                "borderWidth": 1,
                "color": "red",
                "padding": 20,
              },
            ],
          ],
        ]
      }
    >
      Welcome to unit tested app!
    </Text>
  </RCTSafeAreaView>
</View>
`;

exports[`renders correctly B 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "500",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 20,
                "letterSpacing": 0.15,
                "lineHeight": 30,
                "marginVertical": 2,
              },
              Object {
                "backgroundColor": "lightgrey",
                "borderColor": "red",
                "borderWidth": 1,
                "color": "red",
                "padding": 20,
              },
            ],
          ],
        ]
      }
    >
      Welcome to unit tested app!
    </Text>
  </RCTSafeAreaView>
</View>
`;
